src/test/scala/com/cognite/sdk/scala/common/RetryWhile.scala

```diff
@@ -33,4 +33,23 @@ trait RetryWhile {
       })
     result.getOrElse(action)
   }
+
+  def retryWhileEmpty[A](
```

Comment:
That was not that easy - to use the `retryWithExpectedResult` function I somehow have to create the `Assertion`, but all `assert`s and `should` magics are hidden in base classes I do not inherit from. Do you know how this can be reasonably done? I could create another (more general) function that will work with `Boolean`s or `Either`s instead of `Assertion/Exception`s, but duplicating just seemed less tedious...